rfcs/0169-feature-parameter-names.md

```diff
@@ -0,0 +1,572 @@
+---
+feature: feature_parameter_names
+start-date: 2024-01-10
+author: Dmitry Bogatov
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Nixpkgs has mechanisms to exports compile-time options of the upstream build
+system through the named parameters of derivation function, but the names of
+these parameters are inconsistent from package to package and from feature to
+feature.
+
+# Motivation
+[motivation]: #motivation
+
+Developer interface for building environment with specific features enabled and
+disabled is more complicated and and involves more "you just have to know it"
+compared to already existing systems.
+
+1. Knowing that package has a feature parameter (out of knowing the upstream)
+   does not mechanically translates to the name of feature parameter, it can be
+   named in multiple ways.
+
+2. Writing generic code is complicated. Even if list of packages all have
+   optional dependency on package `foo`, duck-typing like
+
+```
+map (p: p.override { with_foo = false; }) packages
+```
+
+won't work because of lack of uniformity, leading ugly code downstream.
```

Comment:
Shouldn't this also work with `.overrideAttrs (old: { cmakeFlags = old.cmakeFlags or [] ++ [ "..." ] })`? Of course this only works when all packages actually have this option.

Reply:
And use the same build system. And name their flags in the same way.

I wish upstreams were to do the work that this RFC tries to do, but it didn't happened in 1991, and it won't happen now.